frontend/test/metabase/scenarios/collections/collections.cy.spec.js

```diff
@@ -477,26 +477,20 @@ describe("scenarios > collection_defaults", () => {
       cy.findByText("First Collection");
     });
 
-    describe("bulk selection (metabase#14705)", () => {
-      it("should be possible to select an item using checkbox", () => {
-        cy.visit("/collection/root");
-        selectItemUsingCheckbox("Orders");
-        cy.findByText("1 item selected").should("be.visible");
-      });
-
-      it("should be possible to select all items at once", () => {
-        cy.icon("dash").click();
-        cy.icon("dash").should("not.exist");
-        cy.findByText("4 items selected");
-      });
-
-      it("should be possible to deselect all items an once", () => {
-        cy.findByTestId("bulk-action-bar").within(() => {
-          cy.icon("check").click();
-        });
-        cy.icon("check").should("not.exist");
-        cy.findByTestId("bulk-action-bar").should("not.be.visible");
+    it("should be possible to apply bulk selection to items (metabase#14705)", () => {
+      cy.visit("/collection/root");
+      selectItemUsingCheckbox("Orders");
+      cy.findByText("1 item selected").should("be.visible");
+      // Select all
+      cy.icon("dash").click();
+      cy.icon("dash").should("not.exist");
+      cy.findByText("4 items selected");
+      // Deselect all
+      cy.findByTestId("bulk-action-bar").within(() => {
+        cy.icon("check").click();
```

Comment:
Did Cypress complain about finding multiple elements and being unable to "apply" click to them? Asking because `frontend/src/metabase/components/StackedCheckBox.jsx` is using 2 checkboxes to achieve this visual effect. I'd argue it is a really bad practice for many reasons - a11y being one of them. That same effect could've been achieved with `::after` or even a carefully crafted shadow in CSS.

Is it worth opening an issue about this and rewriting this component?

Reply:
No, the test is completely green locally. As I see now, `noIcon` prop actually ensures `CheckBox` doesn't render the icon, so Cypress selects only one element here. I'm not a big expert at a11y, so can't say whether it's a big problem or not. However, `noIcon` is used only at `StackedCheckBox` and we should better avoid these kinds of 'exclusive' props.

So, I think it's worth opening an issue. It looks minor, but the devil is in the detail

Reply:
From the a11y prospective, I a bigger problem is that the `CheckBox` is actually a `div` with click handlers instead of `<input type="checkbox" />`

Reply:
Great.
Thank you fro the feedback. Test will work the same even after we fix this component.

Btw, it's much shorter and cleaner now. Looks good 🎉